app/src/main/java/de/danoeh/antennapod/activity/SelectSubscriptionActivity.java

```diff
@@ -0,0 +1,164 @@
+
+    private Disposable disposable;
+    private volatile List<Feed> listItems;
+
+    private SubscriptionSelectionActivityBinding viewBinding;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        setTheme(UserPreferences.getTranslucentTheme());
+        super.onCreate(savedInstanceState);
+
+        viewBinding = SubscriptionSelectionActivityBinding.inflate(getLayoutInflater());
+        setContentView(viewBinding.getRoot());
+        setSupportActionBar(viewBinding.toolbar);
+
+        viewBinding.transparentBackground.setOnClickListener(v -> finish());
+        viewBinding.card.setOnClickListener(null);
+
+        loadSubscriptions();
+
+        final Integer[] checkedPosition = new Integer[1];
+        viewBinding.list.setChoiceMode(ListView.CHOICE_MODE_SINGLE);
+        viewBinding.list.setOnItemClickListener((listView, view1, position, rowId) ->
+                checkedPosition[0] = position
+        );
+        viewBinding.shortcutBtn.setOnClickListener(view -> {
+            if (checkedPosition[0] != null && Intent.ACTION_CREATE_SHORTCUT.equals(
+                    getIntent().getAction())) {
+                getBitmapFromUrl(listItems.get(checkedPosition[0]));
+            }
+        });
+
+    }
+
+    public List<Feed> getFeedItems(List<NavDrawerData.DrawerItem> items, List<Feed> result) {
+        for (NavDrawerData.DrawerItem item : items) {
+            if (item.type == NavDrawerData.DrawerItem.Type.TAG) {
+                getFeedItems(((NavDrawerData.TagDrawerItem) item).children, result);
+            } else {
+                Feed feed = ((NavDrawerData.FeedDrawerItem) item).feed;
+                if (!result.contains(feed)) {
+                    result.add(feed);
+                }
+            }
+        }
+        return result;
+    }
+
+    private void addShortcut(Feed feed, Bitmap bitmap) {
+        if (ShortcutManagerCompat.isRequestPinShortcutSupported(this)) {
+            Intent intent = new Intent(this, MainActivity.class);
+            intent.setAction(Intent.ACTION_MAIN);
+            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+            intent.putExtra(EXTRA_FEED_ID, feed.getId());
+            String id = "subscription-" + feed.getId();
+            IconCompat icon;
+
+            if (bitmap != null) {
+                icon = IconCompat.createWithAdaptiveBitmap(bitmap);
+            } else {
+                icon = IconCompat.createWithResource(this, R.drawable.ic_folder_shortcut);
+            }
+
+            ShortcutInfoCompat shortcut = new ShortcutInfoCompat.Builder(this, id)
+                    .setShortLabel(feed.getTitle())
+                    .setLongLabel(feed.getFeedTitle())
+                    .setIntent(intent)
+                    .setIcon(icon)
+                    .build();
+
+            ShortcutManagerCompat.requestPinShortcut(this, shortcut, null);
```

Comment:
Currently, users have to place the icon twice: They drag the shortcut, then select it from AntennaPod, then get the pin shortcut popup and need to drag it again. Instead of requesting the shortcut, you can also return it to the caller directly. Then users do not need to place the icon twice. That would then also get rid of the "unsupported device" thing below.

Something like this:
```
setResult(Activity.RESULT_OK, ShortcutManager.createShortcutResultIntent(shortcut));
finish();
```

It only works when starting the activity directly instead of from the main activity (see comment above)